### PR TITLE
Fix the dockerfile to run the examples

### PR DESCRIPTION
**Pull request by marcograss**
Fix and optimize the dockerfile and requirements.txt to make the examples runnable

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,7 @@
-FROM python:3.5
+FROM python:3.6
```
Comment (notadamking) [inline Dockerfile]: The framework should support python 3.5, is there something that is not supported?

Comment (dru-zod) [reply Dockerfile]: The Dockerfile i submitted was based on 3.6, wasn't able to get to run with 3.5 I think there's a problem f strings

Comment (notadamking) [reply Dockerfile]: I have updated it to 3.5 and removed all f-strings to fix these issues.

Comment (notadamking) [reply Dockerfile]: I will merge this if you can re-submit w/ python:3.5

Comment (marcograss) [reply Dockerfile]: some type annotations doesn't work on python 3.5 if you run the examples notebooks

Comment (notadamking) [reply Dockerfile]: Working on a fix for the type annotations, for now let's switch this to 3.5.

**Pull request merged**